src/SayMore/Utilities/FileSyncHelper.cs

```diff
@@ -0,0 +1,322 @@
+		{
+			var client = IsSynched(filePath);
+
+			string confirmationString;
+
+			switch (client)
+			{
+				case SyncClient.Dropbox:
+					confirmationString = LocalizationManager.GetString("MainWindow.ConfirmStopDropboxSync",
+						"It looks like this project is in a directory that is being synchronized with Dropbox. This is known to cause problems due to files being locked. Would you like to disable Dropbox temporarily until you close SayMore?");
+					break;
+
+				case SyncClient.GoogleDrive:
+					confirmationString = LocalizationManager.GetString("MainWindow.ConfirmStopGoogleDriveSync",
+						"It looks like this project is in a directory that is being synchronized with Google Drive. This is known to cause problems due to files being locked. Would you like to disable Google Drive temporarily until you close SayMore?");
+					break;
+
+				case SyncClient.OneDrive:
+					confirmationString = LocalizationManager.GetString("MainWindow.ConfirmStopOneDriveSync",
+						"It looks like this project is in a directory that is being synchronized with OneDrive. This is known to cause problems due to files being locked. Would you like to disable OneDrive temporarily until you close SayMore?");
+					break;
+
+				case SyncClient.None:
+					return;
+
+				default:
+					return;
+			}
+
+			if (DialogResult.No == MessageBox.Show(confirmationString, Program.ProductName, MessageBoxButtons.YesNo,
+				    MessageBoxIcon.Question))
+				return;
+
+			StopClient(client);
+		}
+
+		/// <summary>
+		/// Checks if the file is in a directory that is actively being synced (the sync program is running)
+		/// </summary>
+		/// <param name="filePath">Full path to the file being checked</param>
+		/// <returns></returns>
+		public static SyncClient IsSynched(string filePath)
+		{
+			var pathParts = filePath.ToLower().Split(new[] {Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar},
+				StringSplitOptions.RemoveEmptyEntries);
+
+			// Dropbox
+			if (IsThisClient(pathParts, SyncClient.Dropbox))
+				return SyncClient.Dropbox;
+
+			// Google Drive
+			if (IsThisClient(pathParts, SyncClient.GoogleDrive))
+				return SyncClient.GoogleDrive;
+
+			// OneDrive
+			if (IsThisClient(pathParts, SyncClient.OneDrive))
+				return SyncClient.OneDrive;
+
+			return SyncClient.None;
+		}
+
+		/// <summary>
+		/// Stops the selected client and remembers it for when Saymore exits
+		/// </summary>
+		/// <param name="client"></param>
+		public static void StopClient(SyncClient client)
+		{
+			switch (client)
+			{
+				case SyncClient.Dropbox:
+					if (!_stoppedDropbox)
+					{
+						KillProcess(kDropboxProcess);
+						_stoppedDropbox = true;
+					}
+					break;
+
+				case SyncClient.GoogleDrive:
+					if (!_stoppedGoogleDrive)
+					{
+						KillProcess(kGoogleDriveProcess);
+						_stoppedGoogleDrive = true;
+					}
+					break;
+
+				case SyncClient.OneDrive:
+					if (!_stoppedOneDrive)
+					{
+						// OneDrive has an off switch
```

Comment:
This code block is identical to the code for turning it back on, except for the command line parameter. Make a method.